coderd/users_test.go

```diff
@@ -1817,7 +1813,11 @@ func assertPagination(ctx context.Context, t *testing.T, client *codersdk.Client
 		},
 	}))
 	require.NoError(t, err, "first page")
-	require.Equalf(t, page.Users, allUsers[:limit], "first page, limit=%d", limit)
+	for i := range allUsers[:limit] {
+		require.Equalf(t, page.Users[i].Username, allUsers[i].Username, "first page, limit=%d", limit)
+	}
```

Comment:
I think the original comparisons are better since the new comparisons fail at the first instance, and do not show the lists and the list lengths in the failure output. Switching to `assert` feels like it might be spammy.

Can we just make a test helper func like:
```golang
func onlyUsernames[U codersdk.User | database.User](users []U) []string {
	var out []string
	for _, u := range users {
		switch u := (any(u)).(type) {
		case codersdk.User:
			out = append(out, u.Username)
		case database.User:
			out = append(out, u.Username)
		}
	}
	return out
}
```

Then we can bring back the old requires.
```golang
require.Equalf(t, onlyUsernames(page.Users), onlyUsernames(allUsers[:limit]), "first page, limit=%d", limit)
```

Reply:
It's a bit of a hack since I cast to `any` in the function :smile: 

It just gives type safety on the call arguments.